src/components/shared/StackSettings.js

```diff
@@ -136,13 +144,34 @@ class StackSettings extends PureComponent<Props> {
             {this._renderImplementationRadioButton('JavaScript', 'js')}
             {this._renderImplementationRadioButton('Native', 'cpp')}
           </li>
-          {hasJsAllocations && !disableCallTreeSummaryButtons ? (
+          {hasAllocations && !disableCallTreeSummaryButtons ? (
             <li className="stackSettingsListItem stackSettingsFilter">
-              {this._renderCallTreeStrategyRadioButton('Timing', 'timing')}
               {this._renderCallTreeStrategyRadioButton(
-                'JavaScript Allocations',
-                'js-allocations'
+                'Timing',
+                'timing',
+                'Summarize using sampled stacks of executed code over time'
               )}
+              {hasJsAllocations
+                ? this._renderCallTreeStrategyRadioButton(
+                    'JavaScript Allocations',
+                    'js-allocations',
+                    'Summarize using bytes of JavaScript allocated (no de-allocations)'
+                  )
+                : null}
+              {hasNativeAllocations
```

Comment:
There are a lot of combinations here:
* only js allocations
* only native allocations
* both js and antive allocations
* neither one or the other

So here are my questions:
* Do all these possibilities exist?
* Are we asserting all of them in tests?
* Have you manually tested all of them, for example with older profiles?

Reply:
>    Do all these possibilities exist?

Yup. They are all about whether the feature is on or not.

>   Are we asserting all of them in tests?

Yes, each branch is exercised in `src/test/components/ProfileCallTreeView.test.js`

>    Have you manually tested all of them, for example with older profiles?

Yup, I've loaded lots of different things manually in here, with the different combinations.